pages/[slug]/map.js

```diff
@@ -3,32 +3,48 @@ import Header from "@/components/Header";
 import Link from "next/link";
 import Image from "next/image";
 import dynamic from "next/dynamic";
+import { useRouter } from "next/router";
+import Footer from "@/components/Footer";
+import Lottie from "react-lottie-player";
+import lottieJson from "../../public/loadingAnimation.json";
 
-const DynamicMap = dynamic(() => import("../components/Map"), {
+const DynamicMap = dynamic(() => import("@/components/Map"), {
   ssr: false,
 });
 
-export default function MapView({ sights }) {
+export default function MapView({ sights, trips }) {
+  const router = useRouter();
+  const currentTrip = trips.find((trip) => trip.slug === router.query.slug);
+  if (!currentTrip) {
+    return <StyledLottie loop animationData={lottieJson} play />;
+  }
   return (
     <>
-      <Header />
+      <Header trip={currentTrip} />
       <StyledHeading1>Itinerary</StyledHeading1>
-      <ToggleLink href="/itinerary">
+      <ToggleLink href={`/${currentTrip.slug}/itinerary`}>
         <ToggleLinkImage
           src="/list.svg"
           height={40}
           width={40}
           alt="list view"
         />
       </ToggleLink>
-      <DynamicMap sights={sights} />
-      <StyledFooter>
-        <StyledLink href="/">Home</StyledLink>
-      </StyledFooter>
+      <DynamicMap sights={sights} trip={currentTrip} />
+      <Footer buttonlink={`/${currentTrip.slug}`} buttontext="Overview" />
     </>
   );
 }
 
+const StyledLottie = styled(Lottie)`
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 50vw;
+  height: 50vw;
+`;
+
```

Comment:
Do you relate the positioning on the `body` element or do is there a container with `position: relative`? Usually it is better to relate to a relative positioned element to have a better control.